File: src/windows/AmbientContextMcp.Core.Tests/ScopeAliasTests.cs
using AmbientContextMcp.Core.Hub;
using AmbientContextMcp.Core.Models;
using Xunit;

namespace AmbientContextMcp.Core.Tests;

public class ScopeAliasTests
{
    [Fact]
    public void GetStates_with_context_all_returns_all_allowed_states()
    {
        var hub = LocalContextHubTestFactory.CreateInMemory();
        hub.Ingest(new AmbientContextSnapshot
        {
            ObservedAt = DateTimeOffset.UtcNow,
            OutboundStates =
            [
                new() { Name = "presence.bucket", Value = "active", Sensitivity = "low" },
                new() { Name = "foregroundApp.category", Value = "code", Sensitivity = "medium" },
                new() { Name = "media.title", Value = "Imagine", Sensitivity = "high" }
            ]
        });

        var states = hub.GetContextStates(new LocalContextStateRequest
        {
            Scopes = ["context.all:read"]
        });

        Assert.Equal(3, states.States.Count);
    }

    [Fact]
    public void GetStates_with_context_all_equals_high()
    {
        var hub = LocalContextHubTestFactory.CreateInMemory();
        hub.Ingest(new AmbientContextSnapshot
        {
            ObservedAt = DateTimeOffset.UtcNow,
            OutboundStates =
            [
                new() { Name = "presence.bucket", Value = "active", Sensitivity = "low" },
                new() { Name = "media.title", Value = "Imagine", Sensitivity = "high" }
            ]
        });

        var withAll = hub.GetContextStates(new LocalContextStateRequest { Scopes = ["context.all:read"] });
        var withHigh = hub.GetContextStates(new LocalContextStateRequest { Scopes = ["context.high:read"] });

        Assert.Equal(withHigh.States.Count, withAll.States.Count);
    }
}

File: src/macos/Tests/AmbientContextCoreTests/ScopeAliasTests.swift
import Foundation
import Testing
@testable import AmbientContextCore

@Suite("ScopeAlias")
struct ScopeAliasTests {
    @Test("GetStates_with_context_all_returns_all_allowed_states")
    func getStatesWithContextAllReturnsAllAllowedStates() {
        let hub = LocalContextHubTestFactory.createInMemory()
        hub.ingest(AmbientContextSnapshot(
            observedAt: Date(),
            outboundStates: [
                AmbientState(name: "presence.bucket", value: "active", sensitivity: "low"),
                AmbientState(name: "foregroundApp.category", value: "code", sensitivity: "medium"),
                AmbientState(name: "media.title", value: "Imagine", sensitivity: "high")
            ]))

        let states = hub.getContextStates(LocalContextStateRequest(scopes: ["context.all:read"]))

        #expect(states.states.count == 3)
    }

    @Test("GetStates_with_context_all_equals_high")
    func getStatesWithContextAllEqualsHigh() {
        let hub = LocalContextHubTestFactory.createInMemory()
        hub.ingest(AmbientContextSnapshot(
            observedAt: Date(),
            outboundStates: [
                AmbientState(name: "presence.bucket", value: "active", sensitivity: "low"),
                AmbientState(name: "media.title", value: "Imagine", sensitivity: "high")
            ]))

        let withAll = hub.getContextStates(LocalContextStateRequest(scopes: ["context.all:read"]))
        let withHigh = hub.getContextStates(LocalContextStateRequest(scopes: ["context.high:read"]))

        #expect(withAll.states.count == withHigh.states.count)
    }
}
